packages/firebase_crashlytics/firebase_crashlytics/example/lib/main.dart

```diff
@@ -0,0 +1,213 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io';
+
+import 'package:firebase_core/firebase_core.dart';
+import 'package:firebase_crashlytics/firebase_crashlytics.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+
+// Toggle this to cause an async error to be thrown during initialization
+// and to test that runZonedGuarded() catches the error
+final _kShouldTestAsyncErrorOnInit = false;
+
+// Toggle this for testing Crashlytics in your app locally.
+final _kTestingCrashlytics = true;
+
+main() {
+  WidgetsFlutterBinding.ensureInitialized();
+  runZonedGuarded(() {
+    runApp(MyApp());
+  }, (error, stackTrace) {
+    print('runZonedGuarded: Caught error in my root zone.');
+    FirebaseCrashlytics.instance.recordError(error, stackTrace);
+  });
+}
+
+// ignore: public_member_api_docs
+class MyApp extends StatefulWidget {
+  @override
+  _MyAppState createState() => _MyAppState();
+}
+
+class _MyAppState extends State<MyApp> {
+  final GlobalKey<ScaffoldState> _scaffoldKey = GlobalKey<ScaffoldState>();
+
+  // Set default `_initialized` and `_error` state to false
+  bool _initialized = false;
+  bool _error = false;
+
+  void testAsyncErrorOnInit() async {
+    Future<void>.delayed(const Duration(seconds: 2), () {
+      final List<int> list = <int>[];
+      print(list[100]);
+    });
+  }
+
+  // Define an async function to initialize FlutterFire
+  void initializeFlutterFire() async {
+    try {
+      // Wait for Firebase to initialize and set `_initialized` state to true
+      await Firebase.initializeApp();
+
+      if (_kTestingCrashlytics) {
+        // Force enable crashlytics collection enabled if we're testing it.
+        await FirebaseCrashlytics.instance
+            .setCrashlyticsCollectionEnabled(true);
+      } else {
+        // Else only enable it in non-debug builds.
+        // You could additionally extend this to allow users to opt-in.
+        await FirebaseCrashlytics.instance
+            .setCrashlyticsCollectionEnabled(!kDebugMode);
+      }
+
+      // Pass all uncaught errors to Crashlytics.
+      Function originalOnError = FlutterError.onError;
+      FlutterError.onError = (FlutterErrorDetails errorDetails) async {
+        await FirebaseCrashlytics.instance.recordFlutterError(errorDetails);
+        // Forward to original handler.
+        originalOnError(errorDetails);
+      };
+
+      setState(() {
+        _initialized = true;
```

Comment:
`initializeFlutterFire()` calls async code so unlikely to run into this scenario but if it was entirely synchronous, the code could fail when calling `setState` as the widget hasn't been mounted. May be worth putting a guard around the `setState` call  by checking if it's mounted and if it hasn't then could set the value of `_initialized` without wrapping with a `setState()` call

Reply:
@MaikuB would you be willing to send a PR to https://github.com/invertase/flutterfire/tree/%40invertase/crashlytics_v1/packages to improve the example app with this point and all the other points you've mentioned?

Reply:
To clarify, you're referring to a PR for just the changes to the example app right? If so then yeah I can take a look. Will likely need help testing to make sure it's working though last time issues with getting the app to build on my machine. Hopefully have better luck this time...

Reply:
Yes please, just the example app. We have tooling now for the monorepo that should help you get setup locally now without too much difficulty, updated contributing guide is here (wip, so not yet merged): https://github.com/FirebaseExtended/flutterfire/blob/@ehesp/update-contributing/CONTRIBUTING.md

If you're using an IntelliJ based IDE then the tooling will automatically generate run configurations for you so you can just run it from there directly;

![image](https://user-images.githubusercontent.com/5347038/91851405-00ec0880-ec57-11ea-8eb3-7791d698e654.png)

Reply:
Hven't done this before but how I would start making changes based on the fork your team has as I already have a fork based on this repo

Reply:
Actually nevermind, GitHub client lets me branch from this PR :)